### PR TITLE
ci: build images in GitHub Actions

### PR DESCRIPTION
**Pull request by andrzej-stencel**

### CODE REVIEW

File: ci/build-push-x86_64.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+set -exuo pipefail
+
+source "$(dirname "${BASH_SOURCE[0]}")/variables.sh"
+
+docker buildx build \
+    --push \
+    --platform=linux/amd64 \
+    --tag "${REPO_URL}:${BUILD_TAG}-${amd64_arch}" \
+    --file "Dockerfile.${amd64_arch}" \
```
Comment (pmalek-sumo) [inline ci/build-push-x86_64.sh:11]: Where are variables like `amd64_arch` coming from?

File: ci/build-push-arm64v8.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -exuo pipefail
+
+if ! docker buildx ls | grep arm
+then
+    echo "Your Buildx seems to lack ARM architecture support"
```
Comment (pmalek-sumo) [inline ci/build-push-arm64v8.sh:7]: Hmm don't we need to check for particular flavor of ARM? i.e. `v7` `v8` etc ?

File: ci/build-push-x86_64.sh (hunk 1 above)
Comment (andrzej-stencel) [reply ci/build-push-x86_64.sh:11]: From the `variables.sh` script that is being sourced a couple lines above.

File: ci/build-push-arm64v8.sh (hunk 2 above)
Comment (andrzej-stencel) [reply ci/build-push-arm64v8.sh:7]: I guess we could 🤷‍♂️

File: ci/variables.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+set -exuo pipefail
+
+readonly amd64_arch=x86_64
+readonly arm_arch=arm32v7
+readonly arm64_arch=arm64v8
+readonly default_repo_url=public.ecr.aws/sumologic/fluent-bit
+
+REPO_URL=${REPO_URL:-$default_repo_url}
+
+DOT="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
```
Comment (sumo-drosiek) [inline ci/variables.sh:12]: What does mean DOT? Could we use more meaningful name?

File: ci/patch-fluent-bit-version.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+set -exuo pipefail
+
+# Patches the Dockerfiles with the Fluent Bit version retrieved from the build tag.
+# It assumes that the Fluent Bit version is the first part of the argument passed into the function.
+# For example, calling `patch_fluent_bit_version "1.2.3-sumo-x.y.z"` will set Fluent Bit version to 1.2.3.
+function patch_fluent_bit_version() {
+    # Remove everything after first `-sumo` (including), e.g. `1.2.3-sumo-1-rc.1` => `1.2.3`
+    fb_version=${1%%-*}
+    # Check that Fluent Bit version is in correct format (SemVer)
+    [[ "$fb_version" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] || exit 10
```
Comment (sumo-drosiek) [inline ci/patch-fluent-bit-version.sh:12]: `[digits].[digits].[digits]` is not strictly semVer

File: ci/variables.sh (hunk 3 above)
Comment (andrzej-stencel) [reply ci/variables.sh:12]: I spent some time trying to come up with a name that would be both descriptive and short/readable. Apparently I failed 😄 This variable means current directory of the executing script. With the `$DOT` I was going for `$DOT/script.sh` being more-or-less equivalent to `./script.sh`.

Comment (sumo-drosiek) [reply ci/variables.sh:12]: `CI_ROOT`, `CI_DIR` maybe?

File: ci/patch-fluent-bit-version.sh (hunk 4 above)
Comment (sumo-drosiek) [reply ci/patch-fluent-bit-version.sh:12]: I mean it is, but semver is not always only `[digits].[digits].[digits]` :sweat_smile:

**Pull request merged**